### DISCUSSION TITLE
Modifications in "mo_optimization" plugin

### DISCUSSION THREAD

**Discussion by eginer** [QuantumPackage/qp2#341] [General]
@Ydrnan 
Hello Yan, 
I had to make a modification in the mo_optimization plugin to try to adapt it to the casscf_cipsi plugin. 
Essentially what I did is the following : 
a) I split the mo_optimization in to "mo_optimization_utils" and "mo_optimization" 
b) "mo_optimization_utils" contains no executable and all subroutines/providers/EZFIO.cfg that are specific to the optimization but not to the things depending on the providers in "class.irp.f" and the generators/selectors. 
c) "mo_optimization"  contains the executables you created before, and providers needed in class.irp.f together with the specific generators/selectors you needed. 
WARNING : the only BIG CHANGE is that the KEYWORDS are no longer in mo_optimization but in mo_optimization_utils
So you just have to remember that in your scripts. 
Btw, I did not managed to make the orbital optimization work within the CASSCF, if we meet at some point I would like to have your ideas on how to make it work :) 
All the best, 
Manu

**Comment by Ydrnan** [comment]
Hello @eginer ,
I'm going to submit my thesis manuscript at the end of the month.
We can discuss about that by zoom in august or in person in september if you come to Toulouse ;)
Best,
Yann

**Comment by eginer** [comment]
Hi Yann, 
I wish you all the best for your manuscript ! I'm sure it will be great :)
We'll find the time to talk in September I guess !
Best, 
Manu
